sdk/deviceregistry/azure-resourcemanager-deviceregistry/src/test/java/com/azure/resourcemanager/deviceregistry/DeviceRegistryManagerTests.java

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.resourcemanager.deviceregistry;
+
+import com.azure.core.credential.TokenCredential;
+import com.azure.core.http.policy.HttpLogDetailLevel;
+import com.azure.core.http.policy.HttpLogOptions;
+import com.azure.core.management.AzureEnvironment;
+import com.azure.core.management.Region;
+import com.azure.core.management.profile.AzureProfile;
+import com.azure.core.test.TestProxyTestBase;
+import com.azure.core.test.annotation.LiveOnly;
+import com.azure.core.util.Configuration;
+import com.azure.core.util.CoreUtils;
+import com.azure.identity.AzurePowerShellCredentialBuilder;
+import com.azure.resourcemanager.deviceregistry.models.Asset;
+import com.azure.resourcemanager.deviceregistry.models.AssetEndpointProfile;
+import com.azure.resourcemanager.deviceregistry.models.BillingContainer;
+import com.azure.resourcemanager.resources.ResourceManager;
+import com.azure.resourcemanager.resources.fluentcore.policy.ProviderRegistrationPolicy;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+import java.util.List;
+import java.util.Random;
+import java.util.stream.Collectors;
+
+public class DeviceRegistryManagerTests extends TestProxyTestBase {
+    private static final Random RANDOM = new Random();
+    private static final Region REGION = Region.US_EAST;
+    private String resourceGroupName = "rg" + randomPadding();
+    private DeviceRegistryManager deviceRegistryManager = null;
+    private ResourceManager resourceManager;
+    private boolean testEnv;
+
+    @Override
+    public void beforeTest() {
+        final TokenCredential credential = new AzurePowerShellCredentialBuilder().build();
+        final AzureProfile profile = new AzureProfile(AzureEnvironment.AZURE);
+
+        resourceManager = ResourceManager.configure()
+            .withLogOptions(new HttpLogOptions().setLogLevel(HttpLogDetailLevel.BASIC))
+            .authenticate(credential, profile)
+            .withDefaultSubscription();
+
+        String registrationState
+                = resourceManager.providers().getByName("Microsoft.DeviceRegistry").registrationState();
+        if (!"Registered".equals(registrationState) && !"registering".equals(registrationState)) {
+            registrationState = resourceManager.providers().register("Microsoft.DeviceRegistry").registrationState();
+        }
+        while (!"Registered".equals(registrationState) && !"registering".equals(registrationState)) {
+            try {
+                Thread.sleep(5000);
+            } catch (InterruptedException e) {
+                throw new RuntimeException(e);
+            }
+            registrationState = resourceManager.providers().getByName("Microsoft.DeviceRegistry").registrationState();
+        }
```

Comment:
Use the policy.

Reply:
Fixed by [c2c10b6](https://github.com/Azure/azure-sdk-for-java/pull/44383/commits/c2c10b6b23d54ea1821b3ab17b8a3a79461e3e55)